<?xml version="1.0" ?>

<robot name="my_bot" xmlns:xacro="https://www.ros.org/wiki/xacro">

    <material name="blue">
	    <color rgba="0 0 0.8 1"/>
    </material>

    <material name="white">
	    <color rgba="1 1 1 1"/>
    </material>

    <material name="grey">
	    <color rgba="0.5 0.5 0.5 1"/>
    </material>

    <material name="black">
      <color rgba="0.0 0.0 0.0 1.0"/>
    </material>

    <material name="green">
        <color rgba="0.0 0.8 0.0 1.0"/>
    </material>

    <link name="base_link">
        <!--<visual>
          <geometry>
             <box size="0.24 0.24 0.001"/>
          </geometry>
          <material name="white"/>
          <origin rpy="0 0 0" xyz="0 0 0.0005"/>
          <material>
            <color rgba="0 0 0.8 1" />
          </material>
        </visual> -->
        <collision>
          <origin rpy="0 0 0" xyz="0 0 0"/>
          <geometry>
            <box size="0.300 0.300 0.015"/>
          </geometry>
        </collision>
    </link>

    <joint name="base_link_joint" type="fixed">
        <origin xyz="0 0 0.0" rpy="0 0 0" />
        <parent link="base_link" />
        <child link="base_plate" />
    </joint>

    <link name="base_plate">
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://my_bot_new/stl/chassis.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"  />
      <geometry>
        <mesh filename="package://my_bot_new/stl/chassis.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>

    <joint name="right_wheel_joint" type="continuous">
        <origin xyz="0.150 0.100 0.0" rpy="0 0 1.5707963268" />
        <parent link="base_plate" />
        <child link="right_wheel" />
        <axis xyz="1 0 0" />
    </joint>

    <link name="right_wheel">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0 1.5707963268 0" />
      <geometry>
        <mesh filename="package://my_bot_new/stl/wheel.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0 1.5707963268 0" />
      <geometry>
        <mesh filename="package://my_bot_new/stl/wheel.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>

    <joint name="left_wheel_joint" type="continuous">
        <origin xyz="0.150 -0.100 0.0" rpy="0 0 1.5707963268" />
        <parent link="base_plate" />
        <child link="left_wheel" />
        <axis xyz="1 0 0" />
    </joint>

    <link name="left_wheel">
    <visual>
      <origin xyz="-0.035 0.0 0.0" rpy="0 1.5707963268 0" />

      <geometry>
        <mesh filename="package://my_bot_new/stl/wheel.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0 1.5707963268 0" />
      <geometry>
        <mesh filename="package://my_bot_new/stl/wheel.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>

    <!-- joint name="laser_joint" type="fixed">
        <origin xyz="0 0.040 0.090" rpy="0 0 1.5708" />
        <parent link="base_plate" />
        <child link="laser" />
    </joint>

    <link name="laser">
    <visual>
      <origin rpy="0 0 -1.5708" xyz="0.084 -0.242 -0.098"/>
      <geometry>
        <mesh filename="package://mobile_robot_autonomous_navigation/stl/RpLidar.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.242 0.084 -0.098"  />
      <geometry>
        <mesh filename="package://mobile_robot_autonomous_navigation/stl/RpLidar.stl" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
    </link>  -->


</robot>